java/src/edu/wpi/first/wpilibj/networktables/NetworkTable.java

```diff
@@ -864,8 +864,8 @@ else if (value instanceof double[])
       return NetworkTablesJNI.putDoubleArray(pathWithSep + key, (double[])value);
     else if (value instanceof Boolean[])
       return NetworkTablesJNI.putBooleanArray(pathWithSep + key, toNative((Boolean[])value));
-    else if (value instanceof Double[])
-      return NetworkTablesJNI.putDoubleArray(pathWithSep + key, toNative((Double[])value));
+    else if (value instanceof Number[])
```

Comment:
This only works for boxed number arrays (`Integer[]`, `Float[]`, `Double[]`, etc) and fails for the more commonly used primitive arrays (`int[]`, `float[]`, `double[]`, ...)